docs/source/demo.rst

```diff
@@ -0,0 +1,79 @@
+.. Copyright 2016 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+=====
+Demos
+=====
+
+Pure localization
+=================
+
+  .. code-block:: bash
+
+    # Pure localization demo in 2D: We use 2 different 2D bags from the Deutsche
+    # Museum. The first one is used to generate the map, the second to run
+    # pure localization.
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_2d/b2-2016-04-05-14-44-52.bag
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_2d/b2-2016-04-27-12-31-41.bag
+    # Generate the map: Run the next command, wait until cartographer_offline_node finishes.
+    roslaunch cartographer_ros offline_backpack_2d.launch bag_filenames:=${HOME}/Downloads/b2-2016-04-05-14-44-52.bag
+    # Run pure localization:
+    roslaunch cartographer_ros demo_backpack_2d_localization.launch \
+       map_filename:=${HOME}/Downloads/b2-2016-04-05-14-44-52.bag.pbstream \
+       bag_filename:=${HOME}/Downloads/b2-2016-04-27-12-31-41.bag
+
+    # Pure localization demo in 3D: We use 2 different 3D bags from the Deutsche
+    # Museum. The first one is used to generate the map, the second to run
+    # pure localization.
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_3d/b3-2016-04-05-13-54-42.bag
+    wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/backpack_3d/b3-2016-04-05-15-52-20.bag
+    # Generate the map: Run the next command, wait until cartographer_offline_node finishes.
```

Comment:
Assets are not written at the end of offline node anymore, right? Now you get a .pbstream and need to run assets writer. See #577.

Reply:
Correct. This does not write assets, but it writes the `.pbstream` which is all we need. You do not have to run the assets writer for this demo.

Reply:
Okay, but that means that the "Generate the map" part of the command is not accurate.

Reply:
We localize in a map, and that map is in a pbstream file (not an image or other asset). Perhaps the term "map" is overloaded here but I think this is fine.

Reply:
Oh right, sorry, I missed the context.